OnlineGameStore.DAL/Repositories/GenreRepository.cs

```
@@ -36,13 +35,17 @@ public async Task<IEnumerable<Genre>> GetAllAsync()
         catch (DbUpdateException ex)
         {
             Console.WriteLine($"Error adding genre: {ex.Message}");
-            return null;
+        }
+        catch (OperationCanceledException ex)
+        {
+            Console.WriteLine($"Error adding genre: {ex.Message}");
         }
         catch (Exception ex)
         {
             Console.WriteLine($"Unexpected error: {ex.Message}");
-            return null;
         }
+
+        return null;
```

I suggest we prefer throwing exceptions instead of returning null when an operation fails unexpectedly. Returning null can silently hide issues and force the caller to remember to check for null, which often leads to harder-to-debug errors later on.

try
        {
            await _context.Genres.AddAsync(entity, cancellationToken);
            await _context.SaveChangesAsync(cancellationToken);
            return entity;
        }
        catch (DbUpdateException ex)
        {
            Console.WriteLine($"Error updating genre: {ex.Message}");
            throw;
        }

---

I see. 
However, does not throwing the exception only complicate the logic? 
We would then need to catch the exception on the next level. It would only create more cumbersome code, while there is still a possibility of somebody forgetting to catch it and app would crash. 
On this stage, the app does not crash and we would still have the logs